sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -288,14 +299,6 @@ trait FileSourceScanLike extends DataSourceScanExec {
     }
   }
 
-  /**
-   * [[partitionFilters]] can contain subqueries whose results are available only at runtime so
-   * accessing [[selectedPartitions]] should be guarded by this method during planning
-   */
-  private def hasPartitionsAvailableAtRunTime: Boolean = {
```

Comment:
this is not needed anymore because `selectedPartitions` will not evaluate dynamic filters.